github/Organization.py

```diff
@@ -1190,31 +1031,28 @@ def invite_user(
         :param teams: array of :class:`github.Team.Team`
         :rtype: None
         """
-        assert user is github.GithubObject.NotSet or isinstance(user, github.NamedUser.NamedUser), user
-        assert email is github.GithubObject.NotSet or isinstance(email, str), email
-        assert (email is github.GithubObject.NotSet) ^ (
-            user is github.GithubObject.NotSet
-        ), "specify only one of email or user"
-        parameters = {}
-        if user is not github.GithubObject.NotSet:
+        assert is_optional(user, github.NamedUser.NamedUser), user
+        assert is_optional(email, str), email
+        assert is_undefined(email) ^ is_undefined(user), "specify only one of email or user"
```

Comment:
Operator `!=` might be more readable
```suggestion
        assert is_defined(email) != is_defined(user), "specify only one of email or user"
```